### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by Snyssfx**

### CODE REVIEW

File: plugin/action/join/join_test.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ created by example.com/sre/filed/pipeline.(*processor).start
 Isn't panic
 `
 
-func TestJoin(t *testing.T) {
+func TestJoinJoin(t *testing.T) {
```
Comment (vitkovskii) [inline plugin/action/join/join_test.go:154]: why JoinJoin?)

File: plugin/action/throttle/throttle_test.go [hunk 2]
```diff
@@ -106,7 +114,8 @@ func TestSizeThrottle(t *testing.T) {
 	iterations := 5
 
 	totalBuckets := iterations + 1
-	eventsTotal := totalBuckets * (limitA/(len(formats[0])+dateLen) + limitB/(len(formats[1])+dateLen) + defaultLimit/(len(formats[2])+dateLen))
+	test := limitA/(len(formats[0])+dateLen-2) + limitB/(len(formats[1])+dateLen-2) + defaultLimit/(len(formats[2])+dateLen-2)
```
Comment (vitkovskii) [inline plugin/action/throttle/throttle_test.go:117]: why test?

File: plugin/input/file/file.go [hunk 3]
```diff
@@ -155,6 +157,11 @@ type Config struct {
 	//> @maintenance
 	MaintenanceInterval  cfg.Duration `json:"maintenance_interval" default:"10s" parse:"duration"` //*
 	MaintenanceInterval_ time.Duration
+
+	//> @3@4@5@6
+	//>
+	//> It turns on watching for file modifications. Turning it on cause more CPU work, but it is more probable to catch file truncation
+	ShouldWatchModifications bool `json:"should_watch_file_modifications" default:"false"` //*
```
Comment (vitkovskii) [inline plugin/input/file/file.go:164]: `Modifications` => `Changes` ?)) More short

File: plugin/input/http/http_test.go [hunk 4]
```diff
@@ -108,6 +109,9 @@ func TestProcessChunksContinue(t *testing.T) {
 	eventBuff := []byte(`{"a":`)
 	eventBuff = input.processChunk(0, chunk, eventBuff)
 
+	// wait for processors to start.
+	time.Sleep(30 * time.Millisecond)
```
Comment (vitkovskii) [inline plugin/input/http/http_test.go:113]: may we have some event that allows avoid hardcode 30ms?

File: plugin/output/file/file.go [hunk 5]
```diff
@@ -124,6 +121,7 @@ func (p *Plugin) Start(config pipeline.AnyConfig, params *pipeline.OutputPluginP
 	p.idx = p.getStartIdx()
 	p.createNew()
 	p.setNextSealUpTime()
+	logger.Errorf("time setted")
```
Comment (vitkovskii) [inline plugin/output/file/file.go:124]: ?????

File: plugin/action/join/join_test.go (hunk 1 above)
Comment (Snyssfx) [reply plugin/action/join/join_test.go:154]: Чтобы отличать тест от TestJoin в k8s, там тоже переименовал в `TestJoinK8S`

Comment (vitkovskii) [reply plugin/action/join/join_test.go:154]: I would recommend `TestSimpleJoin`/`TestK8SJoin`

**Pull request merged**